conans/test/unittests/util/files/test_dirty.py

```diff
@@ -0,0 +1,38 @@
+# coding=utf-8
+
+import os
+import unittest
+
+
+from conans.test.utils.test_files import temp_folder
+from conans.util.files import set_dirty, clean_dirty, set_dirty_context_manager, _DIRTY_FOLDER
+
+
+class DirtyTest(unittest.TestCase):
+
+    def setUp(self):
+        self.temp_folder = temp_folder()
+        self.dirty_folder = self.temp_folder + _DIRTY_FOLDER
+
+    def test_set_dirty(self):
+        set_dirty(self.temp_folder)
+        self.assertTrue(os.path.exists(self.dirty_folder))
+
+    def test_clean_dirty(self):
+        set_dirty(self.temp_folder)
+        self.assertTrue(os.path.exists(self.dirty_folder))
+        clean_dirty(self.temp_folder)
+        self.assertFalse(os.path.exists(self.dirty_folder))
+
+    def test_set_dirty_context(self):
+        with set_dirty_context_manager(self.temp_folder):
+            self.assertTrue(os.path.exists(self.dirty_folder))
+        self.assertFalse(os.path.exists(self.dirty_folder))
+
+        try:
+            with set_dirty_context_manager(self.temp_folder):
+                self.assertTrue(os.path.exists(self.dirty_folder))
+                self.fail()
+        except AssertionError:
+            pass
```

Comment:
why ignore error?

Reply:
the idea here is reproducing the case where you have an exception in middle of context, breaking it graceful exit. The except clause will catch self.fail() execution, but real test is at line 38, which requires that dirty file is there yet after failing.

Reply:
that's hard to understand by reading code. may be add comment or just use `raise Exception("this is intended to fail for test!")`. otherwise it looks like we are just ignoring failing test

Reply:
also, if `self.assertTrue` fails at the line above, it will be ignored - I don't think it's intended

Reply:
yeah, makes sense, I'll update it

Reply:
done